benchmark/scenarios/test/test_write_request_load.py

```diff
@@ -167,6 +168,39 @@ def assert_created(returned_datasets):
         d2.addCallback(assert_created)
         s.stop()
 
+    @capture_logging(None)
+    def test_cleanup_deletes_dataset(self, _logger):
+        """
+        ``write_request_load_scenario`` deletes the dataset created by
+        the setup when the scenario is stopped.
+        """
+        c = Clock()
+        cluster = self.make_cluster(self.get_fake_flocker_client_instance())
+        s = write_request_load_scenario(c, cluster, 5, sample_size=3)
+
+        def assert_created(returned_datasets):
+            self.assertNotEqual(returned_datasets, [])
+
+        # Create a datasest and verify we get a success
+        d = s.scenario_setup._create_dataset(self.node1)
+        d.addCallback(s.scenario_setup._set_dataset_id)
```

Comment:
This test is too low-level, essentially copying the implementation out of the tested code.

It should ideally run `scenario.start` followed by `scenario.stop`. However, I would also accept `scenario_setup.run_setup` followed by `scenario_setup.run_cleanup`.